src/test/java/apoc/convert/ConvertJsonTest.java

```diff
@@ -235,6 +237,172 @@ public void testToTreeParentNodes () {
+                    List<Map> subParts = (List<Map>)pcParts.get("subcategory");
+                    Map cpu = subParts.get(0);
+                    assertEquals("Category", pcParts.get("_type"));
+                    assertEquals("CPU", cpu.get("name"));
+                    assertFalse("Should not contain key `subcategory.id`",cpu.containsKey("subcategory.id"));
+                    assertEquals("ex",cpu.get("subcategory.subCat"));
+                });
+    }
+
+    @Test
+    public void testToTreeLeafNodesWithConfigOnlyInclude() {
+        statementForConfig(db);
+        String call = "MATCH p=(n:Category)-[:subcategory*]->(m)\n" +
+                "WHERE NOT (m)-[:subcategory]->() AND NOT ()-[:subcategory]->(n)\n" +
+                "WITH COLLECT(p) AS ps\n" +
+                "CALL apoc.convert.toTree(ps, true, {nodes: {Category: ['name', 'surname']}}) yield value\n" +
+                "RETURN value;";
+        testCall(db, call,
+                (row) -> {
+                    Map root = (Map) row.get("value");
+                    assertEquals("Category", root.get("_type"));
+                    assertEquals("PC",root.get("name"));
+                    assertEquals("computer",root.get("surname"));
+                    List<Map> parts = (List<Map>) root.get("subcategory");
+                    assertEquals(1,parts.size());
+                    Map pcParts = parts.get(0);
+                    assertEquals("Category", pcParts.get("_type"));
+                    assertEquals("Parts", pcParts.get("name"));
+                    assertEquals(1L, pcParts.get("subcategory.id"));
+                    assertEquals("gen", pcParts.get("subcategory.subCat"));
+                    List<Map> subParts = (List<Map>)pcParts.get("subcategory");
+                    Map cpu = subParts.get(0);
+                    assertEquals("Category", pcParts.get("_type"));
+                    assertEquals("CPU", cpu.get("name"));
+                    assertEquals(2L, cpu.get("subcategory.id"));
+                    assertEquals("ex",cpu.get("subcategory.subCat"));
+                });
+    }
+
+    @Test
+    public void testToTreeLeafNodesWithConfigErrorInclude() {
+        statementForConfig(db);
+        String call = "MATCH p=(n:Category)-[:subcategory*]->(m)\n" +
+                "WHERE NOT (m)-[:subcategory]->() AND NOT ()-[:subcategory]->(n)\n" +
+                "WITH COLLECT(p) AS ps\n" +
+                "CALL apoc.convert.toTree(ps, true, {nodes: {Category: ['-name','name']}, rels: {subcategory:['-id']}}) yield value\n" +
+                "RETURN value;";
+        try {
+            testResult(db, call, (row) -> { });
+        } catch (QueryExecutionException e) {
+            Throwable except = ExceptionUtils.getRootCause(e);
+            TestCase.assertTrue(except instanceof RuntimeException);
+            assertEquals("Only include or exclude attribute are possible!", except.getMessage());
+            throw e;
+        }
+
+    }
+
+    @Test
```

Comment:
do we need such an elaborate setup just to check the config contains the same type of property filters?